### PR TITLE
Make scan for projects threaded

### LINKED ISSUE #5953: Project manager scan should run in a thread to keep responsive UI
**Operating system or device - Godot version:**
Windows 10, Godot 2.1rc1

**Issue description** (what happened, and what was expected):
Using the scan function at the project manager crashes Godot when I click 'open'.

**Steps to reproduce:**
Open Godot, click Scan, select a large folder, click Open. Shows the Windows Close/Debug error box.

### PR DESCRIPTION
**Pull request by KoBeWi**
Closes #5953

https://github.com/godotengine/godot/assets/2223172/fb60b5b4-7c49-4b55-bc41-a1acc010f1cd

### CODE REVIEW

File: scene/gui/progress_bar.cpp [hunk 1]
```diff
@@ -61,7 +61,7 @@ void ProgressBar::_notification(int p_what) {
 				Size2 size = get_size();
 				real_t fill_size = MIN(size.width, size.height) * 2;
 
-				if (Engine::get_singleton()->is_editor_hint() && !editor_preview_indeterminate) {
+				if (is_part_of_edited_scene() && !editor_preview_indeterminate) {
```
Comment (KoBeWi) [inline scene/gui/progress_bar.cpp:64]: These changes were needed to run indeterminate progress in Project Manager.
Looking at the code, I don't get how it already works in the editor (in template download).

**Review by akien-mga — APPROVED** [review]
Looks good to me, let's merge early on for 4.4.

**Comment by arkology** [comment]
Is this one still good for merging into early 4.4?)

File: editor/project_manager/project_list.cpp [hunk 2]
```diff
@@ -571,25 +622,39 @@ void ProjectList::find_projects(const String &p_path) {
...
+		vb->add_child(label);
 
-	for (const String &E : projects) {
-		add_project(E, false);
+		ProgressBar *progress = memnew(ProgressBar);
+		progress->set_indeterminate(true);
+		vb->add_child(progress);
+
+		add_child(scan_progress);
+		scan_progress->connect("confirmed", callable_mp(this, &ProjectList::_scan_finished));
```
Comment (AThousandShips) [inline editor/project_manager/project_list.cpp:642]:
```suggestion
		scan_progress->connect(SceneStringName(confirmed), callable_mp(this, &ProjectList::_scan_finished));
```

**Review by AThousandShips — APPROVED** [review]

File: editor/project_manager/project_list.cpp [hunk 3]
```diff
@@ -624,25 +675,39 @@ void ProjectList::find_projects(const String &p_path) {
...
 
-	for (const String &E : projects) {
-		add_project(E, false);
+		ProgressBar *progress = memnew(ProgressBar);
+		progress->set_indeterminate(true);
+		vb->add_child(progress);
+
+		add_child(scan_progress);
+		scan_progress->connect(SceneStringName(confirmed), callable_mp(this, &ProjectList::_scan_finished));
+		scan_progress->connect("cancelled", callable_mp(this, &ProjectList::_scan_finished));
```
Comment (arkology) [inline editor/project_manager/project_list.cpp:696]:
```suggestion
		scan_progress->connect("canceled", callable_mp(this, &ProjectList::_scan_finished));
```

**Comment by Repiteo** [comment]
Thanks!

**Pull request merged**